okhttp-tests/src/test/java/okhttp3/HeadersTest.java

```diff
@@ -366,8 +370,14 @@
         .build();
     List<Challenge> challenges = HttpHeaders.parseChallenges(headers, "WWW-Authenticate");
     assertEquals(1, challenges.size());
-    assertEquals("Digest", challenges.get(0).scheme());
```

Comment:
I'm ok with this, but we should mention this as a potentially breaking change.  I hope most people using challenges are already using them case insensitively.

e.g. this will break

https://github.com/rburgst/okhttp-digest/blob/0205d0ef00273ce800bb012d4fc1cfc72857a3bc/src/test/java/com/burgstaller/okhttp/WrongOrderedAuthenticationHeaderTest.java#L27

Reply:
I’m curious what the benefit of forcing lowercase is? It costs an allocation.

Reply:
I like this, but it's whether we want a small amount of upgrade pain for what is probably a small set of users.  I'm not as concerned about the allocation, nor turkish i characters :)

Reply:
I'd hope most users stick to the spec and compare case-insentively, so shouldn't hit too many I hope